aws/table_aws_cloudwatch_metric_statistic_data_point.go

```diff
@@ -0,0 +1,214 @@
+			},
+			{
+				Name:        "dimensions",
+				Description: "The dimensions for the metric.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Label"),
+			},
+		})),
+	}
+}
+
+type MetricStatistics struct {
+	MetricName *string
+	Namespace  *string
+	Period     *int32
+	Label      *string
+	Dimensions []types.Dimension
+	types.Datapoint
+}
+
+//// LIST FUNCTION
+
+func listCloudWatchMetricStatisticDataPoints(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// set the input parameters
+	params := &cloudwatch.GetMetricStatisticsInput{
+		MetricName: aws.String(d.EqualsQuals["metric_name"].GetStringValue()),
+		Namespace:  aws.String(d.EqualsQuals["namespace"].GetStringValue()),
+		Statistics: types.Statistic.Values(types.StatisticMaximum),
+	}
+
+	//set the start and end time based on the provided timestamp
+	if d.Quals["timestamp"] != nil {
+		for _, q := range d.Quals["timestamp"].Quals {
+			timestamp := q.Value.GetTimestampValue().AsTime()
+			switch q.Operator {
+			case "=":
+				params.StartTime = aws.Time(timestamp)
+				params.EndTime = aws.Time(timestamp)
+			case ">=", ">":
+				params.StartTime = aws.Time(timestamp)
+			case "<", "<=":
+				params.EndTime = aws.Time(timestamp)
+			}
+		}
+	} else {
+		params.StartTime = aws.Time(time.Now().AddDate(0, 0, -1))
+		params.EndTime = aws.Time(time.Now())
+	}
+
+	if params.StartTime == nil {
+		params.StartTime = aws.Time(time.Now().AddDate(0, 0, -1))
+	}
+	if params.EndTime == nil {
+		params.EndTime = aws.Time(time.Now())
+	}
+
+	// set the period based on the duration between the start and end time
+	// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudwatch@v1.25.1#GetMetricStatisticsInput.Period
+	// here we have tried setting the period in such a way that it could provide a good spread under 1440 datapoints
+
+	// for an example for a 5 days duration the maximum datapoints could be (5 * 24 * 3600) = 432000
+	// now due to API limitation of 1440, as per the below calculation, period will be 432000/1440 = 300 and with this period we will get upto 1440 datapoints
+
+	// another example, for a 5 days 15 hours duration the maximum datapoints could be (5 * 24 + 15) * 3600) = 486000
+	// now due to API limitation of 1440, as per the below calculation, period will be ((486000/1440)/60 + 1)*60 = 360
+	// in this case 486000/1440 = 337, which is not multiple of 60, so the closest multiple of 60 after 337 is 360
+	// with this period we will get upto 1350 datapoints
+
+	duration := params.EndTime.Sub(*params.StartTime).Hours()
+	durationSec := int32(duration) * 3600
+	defaultPeriod := (int32(duration) * 3600) / 1440
+
+	// if the duration is less than 3 hrs
+	if duration < 3 {
+		params.Period = aws.Int32(10)
```

Comment:
Why did we pick 10 seconds if under 3 hours? I couldn't find anything in AWS' docs, but I probably missed something

Reply:
updated the logic, default minimum is 60 sec